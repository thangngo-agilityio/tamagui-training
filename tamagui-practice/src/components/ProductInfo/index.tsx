'use client';

import {
  Box,
  Button,
  Flex,
  Grid,
  GridItem,
  Heading,
  HStack,
  Stack,
  Text,
  VStack,
} from '@chakra-ui/react';
import { useRouter } from 'next/navigation';
import dynamic from 'next/dynamic';
import { Fragment, useCallback, useState } from 'react';

// Actions
import { updateMyCart } from '@/actions';

// Hooks
import { useCustomToast } from '@/hooks';

// Utils
import { formatAmountNumber } from '@/utils';

// Icons
import { MinusIcon, PlusIcon } from '@/icons';

// Constants
import { BENEFIT_LIST, ROUTER, SUCCESS_MESSAGES } from '@/constants';

// Types
import { ICartItem, TProduct } from '@/types';

// Components
const ImageProduct = dynamic(() => import('@/components/ImageProduct'));
const CardBenefit = dynamic(() => import('@/components/CardBenefit'));

type TProductInfo = {
  cartId: string;
  cartItems: ICartItem[];
  product?: TProduct;
  length?: string;
  width?: string;
};

const ProductInfo = ({ product, cartId, cartItems }: TProductInfo) => {
  const [quantity, setQuantity] = useState(1);
  const { showToast } = useCustomToast();
  const router = useRouter();

  const {
    id: productId = '',
    image = [],
    name = '',
    price = 0,
    description = '',
  } = product || {};

  const handleAddToCart = useCallback(async () => {
    const newCartItems = [...cartItems];

    const itemExist = newCartItems.find((cartItem) => {
      const { product: productItem } = cartItem || {};
      const { id } = productItem || {};

      return id === productId;
    });

    if (itemExist) {
      itemExist.quantity = quantity;
    } else {
      newCartItems.push({ product, quantity: quantity });
    }

    const res = await updateMyCart(cartId, newCartItems);

    const { error } = res || {};

    if (error) {
      showToast(error);
    } else {
      showToast(SUCCESS_MESSAGES.ADD_CART, 'success');
    }
  }, [cartId, cartItems, product, productId, quantity, showToast]);

  const handleBuyNow = useCallback(async () => {
    handleAddToCart();

    return router.push(ROUTER.CART);
  }, [handleAddToCart, router]);

  // Handle change quantity
  const onChange = useCallback(
    (operation: 'increment' | 'decrement') => {
      setQuantity((prevQuantity) =>
        operation === 'increment' ? prevQuantity + 1 : prevQuantity - 1,
      );
    },
    [setQuantity],
  );

  const handleDecrement = useCallback(async () => {
    onChange('decrement');
  }, [onChange]);

  const handleIncrement = useCallback(async () => {
    onChange('increment');
  }, [onChange]);

  return (
    <Flex flexDir={{ base: 'column', lg: 'row' }} gap="42px">
      <ImageProduct image={image} alt={name} />
      <Stack flexDir="column" flex={1}>
        <Heading
          size={{ base: 'size4xl', lg: 'size6xl' }}
          variant="quinary"
          mb="5px"
        >
          {name}
        </Heading>
        <Text
          size={{ base: 'textMd', lg: 'text2Xl' }}
          variant="senary"
          mb={{ base: '10px', lg: '30px' }}
        >
          {description}
        </Text>
        <VStack mb={{ base: '10px', lg: '30px' }} alignItems="flex-start">
          <Heading variant="productTitle" size="size2xl">
            Dimension:
          </Heading>
          <Flex flexDir="row" gap="5px">
            <Text variant="senary" size={{ base: 'textMd', lg: 'text2Xl' }}>
              Length-34cm,
            </Text>
            <Text variant="tertiary" size={{ base: 'textMd', lg: 'text2Xl' }}>
              Width-56cm
            </Text>
          </Flex>
        </VStack>

        <Box
          borderTopWidth="1px"
          borderBottomWidth="1px"
          borderColor="border.300"
          mb="28px"
        >
          <Heading size="size9xl" variant="quinary" py="10px">
            N{formatAmountNumber(price?.toString())}
          </Heading>
        </Box>

        <Stack flexDir="row" mb="44px" gap="60px">
          <VStack alignItems="flex-start">
            <Heading variant="quaternary" size="lg" mb="20px">
              Quantity available
            </Heading>
            <HStack gap="20px">
              <Button
                variant="quantity"
                size="quantity"
                data-testid="btn-decrement"
                onClick={handleDecrement}
                isDisabled={quantity === 1}
                aria-label="Minus button"
              >
                <MinusIcon />
              </Button>
              <Text variant="quantity">{quantity}</Text>
              <Button
                variant="quantity"
                size="quantity"
                onClick={handleIncrement}
                data-testid="btn-increment"
                aria-label="Plus button"
              >
                <PlusIcon />
              </Button>
            </HStack>
          </VStack>
          <VStack alignItems="flex-start">
            <Heading mb="20px" variant="quaternary" size="lg">
              Color:
            </Heading>
            <HStack>
              <Box
                w="36px"
                h="36px"
                borderRadius="100%"
                bgColor="background.1400"
              />
              <Box
                w="36px"
                h="36px"
                borderRadius="100%"
                bgColor="background.1500"
              />
              <Box
                w="36px"
                h="36px"
                borderRadius="100%"
                bgColor="background.1600"
              />
            </HStack>
          </VStack>
        </Stack>

        <Grid
          flexDirection="row"
          gap={{ base: '20px', lg: '92px' }}
          mb={{ base: '20px', lg: '40px' }}
          templateColumns={{ base: '', lg: 'repeat(2, 1fr)' }}
        >
          {BENEFIT_LIST.map((item) => {
            const IconComponent = item.icon || Fragment;
            return (
              <GridItem key={item.id}>
                <CardBenefit
                  icon={<IconComponent />}
                  title={item.title}
                  text={item.text}
                />
              </GridItem>
            );
          })}
        </Grid>

        <Heading
          variant="quinary"
          size="size2xl"
          mb={{ base: '25px', lg: '58px' }}
        >
          Delivery Fee: N3,000-N5000
        </Heading>

        <Stack
          flexDir={{ base: 'column', lg: 'row' }}
          gap={{ base: '20px', lg: 'unset' }}
        >
          <Button
            size={{ base: 'xl', lg: 'productDetail' }}
            variant="buy"
            onClick={handleBuyNow}
          >
            Buy now
          </Button>
          <Button
            size={{ base: 'xl', lg: 'productDetail' }}
            variant="cart"
            ml={{ base: 'unset', lg: '30px' }}
            onClick={handleAddToCart}
          >
            Add to cart
          </Button>
        </Stack>
      </Stack>
    </Flex>
  );
};

export default ProductInfo;
