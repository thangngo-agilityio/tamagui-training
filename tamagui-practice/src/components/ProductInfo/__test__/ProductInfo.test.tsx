import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import ProductInfo from '../index';
import '@testing-library/jest-dom';

// Mocking formatAmountNumber utility
jest.mock('@/utils', () => ({
  formatAmountNumber: jest.fn((price: string) => `formatted-${price}`),
}));

describe('ProductInfo() ProductInfo method', () => {
  const mockOnClickBuy = jest.fn();
  const mockOnClickAddCard = jest.fn();

  const defaultProps = {
    title: 'Test Product',
    price: 1000,
    description: 'Test Description',
    onClickBuy: mockOnClickBuy,
    onClickAddCard: mockOnClickAddCard,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot for ProductInfo', () => {
    const { container } = render(<ProductInfo {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  describe('Happy Path', () => {
    it('should render the product title, description, and formatted price', () => {
      render(<ProductInfo {...defaultProps} />);

      expect(screen.getByText('Test Product')).toBeInTheDocument();
      expect(screen.getByText('Test Description')).toBeInTheDocument();
      expect(screen.getByText('Nformatted-1000')).toBeInTheDocument();
    });

    it('should increment quantity when increment button is clicked', () => {
      render(<ProductInfo {...defaultProps} />);

      const incrementButton = screen.getByTestId('btn-increment');
      fireEvent.click(incrementButton);

      expect(screen.getByText('2')).toBeInTheDocument();
    });

    it('should decrement quantity when decrement button is clicked', () => {
      render(<ProductInfo {...defaultProps} />);

      const incrementButton = screen.getByTestId('btn-increment');
      fireEvent.click(incrementButton); // Increment to 2

      const decrementButton = screen.getByTestId('btn-decrement');
      fireEvent.click(decrementButton); // Decrement back to 1

      expect(screen.getByText('1')).toBeInTheDocument();
    });

    it('should call onClickBuy when Buy now button is clicked', () => {
      render(<ProductInfo {...defaultProps} />);

      const buyButton = screen.getByText('Buy now');
      fireEvent.click(buyButton);

      expect(mockOnClickBuy).toHaveBeenCalled();
    });

    it('should call onClickAddCard when Add to cart button is clicked', () => {
      render(<ProductInfo {...defaultProps} />);

      const addToCartButton = screen.getByText('Add to cart');
      fireEvent.click(addToCartButton);

      expect(mockOnClickAddCard).toHaveBeenCalled();
    });
  });

  describe('Edge Cases', () => {
    it('should disable decrement button when quantity is 1', () => {
      render(<ProductInfo {...defaultProps} />);

      const decrementButton = screen.getByTestId('btn-decrement');
      expect(decrementButton).toBeDisabled();
    });

    it('should handle missing description gracefully', () => {
      const propsWithoutDescription = {
        ...defaultProps,
        description: undefined,
      };
      render(<ProductInfo {...propsWithoutDescription} />);

      expect(screen.queryByText('Test Description')).not.toBeInTheDocument();
    });

    it('should handle missing price gracefully', () => {
      const propsWithoutPrice = { ...defaultProps, price: 0 };
      render(<ProductInfo {...propsWithoutPrice} />);

      expect(screen.queryByText('Nformatted-')).not.toBeInTheDocument();
    });
  });
});
