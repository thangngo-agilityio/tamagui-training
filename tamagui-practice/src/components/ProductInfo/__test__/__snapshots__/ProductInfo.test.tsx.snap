// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfo() ProductInfo method should match snapshot for ProductInfo 1`] = `
<div>
  <div
    class="chakra-stack css-1bef4uc"
  >
    <h2
      class="chakra-heading css-r7q7qr"
    >
      Test Product
    </h2>
    <p
      class="chakra-text css-ns37wz"
    >
      Test Description
    </p>
    <div
      class="chakra-stack css-13co5ed"
    >
      <h2
        class="chakra-heading css-0"
      >
        Dimension:
      </h2>
      <div
        class="css-tx8ton"
      >
        <p
          class="chakra-text css-0"
        >
          Length-34cm,
        </p>
        <p
          class="chakra-text css-0"
        >
          Width-56cm
        </p>
      </div>
    </div>
    <div
      class="css-ucr90f"
    >
      <h2
        class="chakra-heading css-9452eu"
      >
        N
        formatted-1000
      </h2>
    </div>
    <div
      class="chakra-stack css-1bsujku"
    >
      <div
        class="chakra-stack css-1ey9w6j"
      >
        <h2
          class="chakra-heading css-uewl2b"
        >
          Quantity available
        </h2>
        <div
          class="chakra-stack css-1qc97x9"
        >
          <button
            aria-label="Minus button"
            class="chakra-button css-4xx2wk"
            data-testid="btn-decrement"
            type="button"
          >
            <svg
              fill="none"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#6E70FF"
                height="10.8411"
                rx="0.864"
                transform="rotate(-90 0.514893 6.6543)"
                width="1.728"
                x="0.514893"
                y="6.6543"
              />
            </svg>
          </button>
          <p
            class="chakra-text css-0"
          />
          <button
            aria-label="Plus button"
            class="chakra-button css-4xx2wk"
            data-testid="btn-increment"
            type="button"
          >
            <svg
              fill="none"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.91666 11.1101C6.39565 11.1101 6.78395 10.7218 6.78395 10.2428V6.49551H10.5518C10.9856 6.49551 11.3372 6.14385 11.3372 5.71006C11.3372 5.27626 10.9856 4.9246 10.5518 4.9246H6.78395V1.17735C6.78395 0.698358 6.39565 0.310059 5.91666 0.310059C5.43767 0.310059 5.04937 0.698358 5.04937 1.17735V4.9246H1.28155C0.847754 4.9246 0.496094 5.27626 0.496094 5.71006C0.496094 6.14385 0.847754 6.49551 1.28155 6.49551H5.04937V10.2428C5.04937 10.7218 5.43767 11.1101 5.91666 11.1101Z"
                fill="#6E70FF"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="chakra-stack css-1ey9w6j"
      >
        <h2
          class="chakra-heading css-uewl2b"
        >
          Color:
        </h2>
        <div
          class="chakra-stack css-1igwmid"
        >
          <div
            class="css-nziuf7"
          />
          <div
            class="css-14xjfk7"
          />
          <div
            class="css-adyymn"
          />
        </div>
      </div>
    </div>
    <div
      class="css-q6rwhn"
    >
      <div
        class="css-1gvp5h0"
      >
        <div
          class="css-1p1m4ay"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32.903 2.34375C32.0495 2.34375 31.2494 2.57414 30.5593 2.97437V1.17188C30.5593 0.524687 30.0346 0 29.3874 0H10.6374C9.99016 0 9.46547 0.524687 9.46547 1.17188V2.98867C8.76992 2.57953 7.96063 2.34375 7.09695 2.34375C4.51227 2.34375 2.40945 4.44656 2.40945 7.03125C2.40945 9.00273 2.99367 10.9077 4.09891 12.5402C5.98047 15.3193 8.3007 16.0405 10.2206 16.8084C11.3277 19.5653 13.5663 21.7502 16.3589 22.7846L15.4948 28.2812H15.3249C13.3864 28.2812 11.8093 29.8584 11.8093 31.7969V37.6562H10.6374C9.99023 37.6562 9.46555 38.1809 9.46555 38.8281C9.46555 39.4753 9.99023 40 10.6374 40H29.3874C30.0346 40 30.5593 39.4753 30.5593 38.8281C30.5593 38.1809 30.0346 37.6562 29.3874 37.6562H28.2155V31.7969C28.2155 29.8584 26.6384 28.2812 24.6999 28.2812H24.5301L23.666 22.7847C26.4627 21.7487 28.7038 19.5591 29.8092 16.7966C31.6229 16.0711 34.0003 15.3478 35.9012 12.5402C37.0064 10.9077 37.5906 9.00266 37.5906 7.03125C37.5905 4.44656 35.4877 2.34375 32.903 2.34375ZM9.4568 13.9787C6.59945 12.8357 4.7532 10.1087 4.7532 7.03125C4.7532 5.73891 5.80461 4.6875 7.09695 4.6875C8.3893 4.6875 9.4407 5.73891 9.4407 7.03125C9.4407 7.11367 9.44938 7.19398 9.46555 7.27156V12.8906C9.46555 13.2673 9.48578 13.6392 9.52453 14.0058L9.4568 13.9787ZM25.8718 37.6562H14.153V35.3125H25.8718V37.6562ZM24.6999 30.625C25.3461 30.625 25.8718 31.1507 25.8718 31.7969V32.9688H14.153V31.7969C14.153 31.1507 14.6788 30.625 15.3249 30.625C15.9607 30.625 23.2089 30.625 24.6999 30.625ZM17.8673 28.2812L18.6427 23.3487C19.0912 23.4071 19.5483 23.4375 20.0124 23.4375C20.4766 23.4375 20.9337 23.407 21.3821 23.3487L22.1576 28.2812H17.8673ZM28.2155 12.8906C28.2155 17.4138 24.5356 21.0938 20.0124 21.0938C15.4892 21.0938 11.8093 17.4138 11.8093 12.8906V7.03125H28.2155V12.8906ZM28.2155 4.6875H11.8093V2.34375H28.2155V4.6875ZM30.5431 13.9787L30.5015 13.9953C30.5395 13.6322 30.5592 13.2637 30.5592 12.8906V7.03125C30.5593 5.73891 31.6107 4.6875 32.903 4.6875C34.1954 4.6875 35.2468 5.73891 35.2468 7.03125C35.2468 10.1087 33.4005 12.8358 30.5431 13.9787Z"
                fill="#3A3A3A"
              />
            </svg>
          </div>
          <div
            class="css-vfucg9"
          >
            <h2
              class="chakra-heading css-wlea3r"
            >
              High Quality
            </h2>
            <p
              class="chakra-text css-0"
            >
              crafted from top materials
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-1gvp5h0"
      >
        <div
          class="css-1p1m4ay"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_65_583)"
              >
                <path
                  d="M37.8573 14.1742C37.3295 12.5398 37.5545 10.0864 36.1827 8.19214C34.8 6.28292 32.3952 5.74026 31.0373 4.74714C29.6938 3.76448 28.4438 1.62612 26.1837 0.88862C23.9873 0.171901 21.7414 1.14448 20 1.14448C18.2588 1.14448 16.013 0.171667 13.8163 0.888542C11.5566 1.62589 10.3056 3.76471 8.96289 4.7469C7.60648 5.73885 5.19992 6.283 3.8175 8.1919C2.4468 10.0846 2.66953 12.5436 2.14266 14.1741C1.64125 15.7259 0 17.5883 0 20.0002C0 22.4135 1.63938 24.2688 2.14266 25.8262C2.67047 27.4606 2.44547 29.9139 3.81734 31.8082C5.19992 33.7175 7.60461 34.2601 8.96266 35.2533C10.306 36.2358 11.5562 38.3743 13.8163 39.1117C16.0112 39.828 18.2606 38.8559 20 38.8559C21.737 38.8559 23.9916 39.827 26.1837 39.1118C28.4434 38.3745 29.6937 36.2361 31.0371 35.2535C32.3935 34.2615 34.8001 33.7174 36.1825 31.8085C37.5533 29.9157 37.3304 27.4569 37.8573 25.8263C38.3587 24.2744 40 22.412 40 20.0002C40 17.587 38.361 15.7323 37.8573 14.1742ZM34.8838 24.8653C34.2686 26.7696 34.4298 28.9008 33.6516 29.9753C32.863 31.0642 30.7913 31.5615 29.1926 32.7309C27.6114 33.8872 26.5032 35.7203 25.2143 36.1408C23.9949 36.5388 22.0077 35.7307 20.0001 35.7307C17.9777 35.7307 16.011 36.5405 14.7858 36.1408C13.497 35.7203 12.3904 33.8885 10.8075 32.7308C9.21828 31.5685 7.13469 31.061 6.34836 29.9752C5.57273 28.9042 5.72781 26.7583 5.11633 24.8654C4.51695 23.0108 3.125 21.4049 3.125 20.0002C3.125 18.594 4.5157 16.9937 5.11617 15.135C5.73141 13.2308 5.57023 11.0995 6.34836 10.025C7.13648 8.9369 9.20984 8.43792 10.8075 7.26948C12.3937 6.1094 13.4949 4.28073 14.7856 3.85956C16.004 3.46206 17.9977 4.26964 19.9999 4.26964C22.0259 4.26964 23.9877 3.45932 25.2142 3.85956C26.5028 4.28003 27.6104 6.11253 29.1926 7.26956C30.7816 8.43182 32.8653 8.9394 33.6516 10.0251C34.4274 11.0963 34.2715 13.24 34.8837 15.1349V15.1349C35.483 16.9896 36.875 18.5954 36.875 20.0002C36.875 21.4064 35.4843 23.0067 34.8838 24.8653ZM27.1987 14.9849C27.8089 15.5952 27.8089 16.5845 27.1987 17.1946L19.3779 25.0154C18.7677 25.6257 17.7783 25.6256 17.1681 25.0154L12.8014 20.6487C12.1912 20.0385 12.1911 19.0492 12.8013 18.439C13.4116 17.8289 14.401 17.8288 15.011 18.439L18.273 21.7009L24.9888 14.985C25.5991 14.3748 26.5884 14.3748 27.1987 14.9849Z"
                  fill="#3A3A3A"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_65_583"
                >
                  <rect
                    fill="white"
                    height="40"
                    width="40"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="css-vfucg9"
          >
            <h2
              class="chakra-heading css-wlea3r"
            >
              Warrany Protection
            </h2>
            <p
              class="chakra-text css-0"
            >
              Over 8 Months
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-1gvp5h0"
      >
        <div
          class="css-1p1m4ay"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_65_593)"
              >
                <path
                  d="M36.8986 20.7348V2.51992C36.8986 1.87009 36.3718 1.34326 35.722 1.34326H4.19674C3.54691 1.34326 3.02008 1.87009 3.02008 2.51992V22.1868C2.45246 22.2695 1.89755 22.4899 1.40664 22.8543C-0.140899 23.9517 -0.506918 26.4074 0.789368 27.922L6.08928 34.3471C9.36996 38.1867 12.855 38.6571 18.0359 38.6571C22.4819 38.6571 24.4696 38.6767 28.3443 37.7938L32.1173 36.8914C32.7291 37.7457 33.7003 38.2992 34.7922 38.2992H36.6508C38.4975 38.2992 40 36.7164 40 34.7709V24.2524C40.0001 22.3948 38.63 20.8689 36.8986 20.7348ZM31.6098 23.1552L30.042 22.359C27.4642 21.0578 24.4839 21.0096 21.8655 22.227C21.1511 22.508 19.5929 23.4507 18.7847 23.4192H13.3445C11.54 23.4192 10.0718 24.8873 10.0718 26.6919V27.55C10.0617 27.5393 10.0511 27.5292 10.0411 27.5184L6.15619 23.3015C5.9211 23.0463 5.65697 22.8328 5.37348 22.6614V10.9875H15.0027V15.3008C15.0027 15.9506 15.5295 16.4775 16.1794 16.4775H23.6042C24.254 16.4775 24.7808 15.9506 24.7808 15.3008V10.9875H34.5454V20.7347C33.1706 20.8409 32.0242 21.8244 31.6098 23.1552ZM17.356 10.9875H22.4274V14.1241H17.356V10.9875ZM34.5453 8.63416H24.7808V3.69658H34.5454V8.63416H34.5453ZM22.4275 3.69658V8.63424H17.356V3.69658H22.4275ZM15.0026 3.69658V8.63424H5.3734V3.69658H15.0026ZM27.7982 35.5047C24.2188 36.3245 22.0631 36.2913 18.0781 36.2913C13.2146 36.2913 10.8646 36.2127 7.90479 32.8495L2.60488 26.4244C1.683 25.224 3.35276 23.8069 4.4254 24.8958L8.31034 29.1128C9.3371 30.1962 10.6291 30.7925 12.2216 30.8275H22.9926C23.6425 30.8275 24.1693 30.3007 24.1693 29.6509C24.1693 29.001 23.6425 28.4742 22.9926 28.4742H12.425V26.6917C12.425 26.1848 12.8374 25.7724 13.3444 25.7724H18.7847C20.0103 25.8601 21.7793 24.8368 22.8575 24.3608C24.8186 23.449 27.0508 23.4851 28.9789 24.4583L31.4432 25.7099V34.6328L27.7982 35.5047ZM37.6467 34.7708C37.6467 35.4187 37.2 35.9458 36.6509 35.9458H34.7922C34.2431 35.9458 33.7965 35.4187 33.7965 34.7708V24.2524C33.7965 23.6046 34.2432 23.0774 34.7922 23.0774H36.6509C37.2 23.0774 37.6467 23.6045 37.6467 24.2524V34.7708Z"
                  fill="#3A3A3A"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_65_593"
                >
                  <rect
                    fill="white"
                    height="40"
                    width="40"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="css-vfucg9"
          >
            <h2
              class="chakra-heading css-wlea3r"
            >
              Free Shipping
            </h2>
            <p
              class="chakra-text css-0"
            >
              Order over N300,000
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-1gvp5h0"
      >
        <div
          class="css-1p1m4ay"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_65_600)"
              >
                <path
                  d="M36.4624 15.9752C36.1995 7.12165 28.9161 0 20 0C11.0838 0 3.80043 7.12165 3.53745 15.9752L2.35291 17.1598V25.1932L3.52938 26.3696V31.8431C3.52938 35.0435 6.13298 37.6471 9.3333 37.6471H10.7906C11.2762 39.0164 12.5839 40 14.1176 40H16.4706C18.0053 40 19.3137 39.015 19.7986 37.6442C19.8658 37.6454 19.9331 37.6471 20 37.6471C25.6814 37.6471 30.4351 33.5987 31.5281 28.2353H34.6049L37.647 25.1932V17.1598L36.4624 15.9752ZM20 2.35294C26.9828 2.35294 32.7969 7.44933 33.9204 14.1176H31.5281C30.4351 8.7542 25.6814 4.70588 20 4.70588C14.3186 4.70588 9.56483 8.7542 8.47181 14.1176H6.07949C7.20302 7.44933 13.0171 2.35294 20 2.35294ZM29.1139 14.1176C25.3109 14.1063 22.6246 14.3379 20.4098 11.1004L19.3344 9.52839L18.4102 11.1938C17.0635 13.6204 15.2499 14.1176 12.9411 14.1176H10.886C11.9335 10.0629 15.6225 7.05882 20 7.05882C24.3775 7.05882 28.0665 10.0629 29.1139 14.1176ZM8.23526 25.8824H6.36961L4.70585 24.2186V18.1344L6.36961 16.4706H8.23526V25.8824ZM5.88232 31.8431V28.2353H8.46953C8.91244 30.4294 9.97212 32.4435 11.542 34.0611C11.2129 34.4127 10.9548 34.8311 10.7906 35.2941H9.3333C7.4304 35.2941 5.88232 33.746 5.88232 31.8431ZM16.4706 37.6471H14.1176C13.4688 37.6471 12.9411 37.1194 12.9411 36.4706C12.9411 35.8219 13.4688 35.2941 14.1176 35.2941H16.4706C17.1193 35.2941 17.647 35.8219 17.647 36.4706C17.647 37.1194 17.1193 37.6471 16.4706 37.6471ZM29.4117 25.8824C29.4117 31.072 25.1896 35.2941 20 35.2941C19.9325 35.2941 19.8645 35.2928 19.7966 35.2912C19.3102 33.9234 18.0032 32.9412 16.4706 32.9412C14.0549 32.9465 14.0934 32.9291 13.7923 32.9568C11.7498 31.1665 10.5882 28.6166 10.5882 25.8824V16.4706H12.9411C15.0233 16.4706 17.5589 16.1273 19.5218 13.7384C22.1116 16.4524 25.2764 16.4706 28.1175 16.4706H29.4117V25.8824ZM35.2941 24.2186L33.6303 25.8824H31.7647V16.4706H33.6303L35.2941 18.1344V24.2186Z"
                  fill="#3A3A3A"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_65_600"
                >
                  <rect
                    fill="white"
                    height="40"
                    width="40"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="css-vfucg9"
          >
            <h2
              class="chakra-heading css-wlea3r"
            >
              24 / 7 Support
            </h2>
            <p
              class="chakra-text css-0"
            >
              Dedicated support
            </p>
          </div>
        </div>
      </div>
    </div>
    <h2
      class="chakra-heading css-1ie70rz"
    >
      Delivery Fee: N3,000-N5000
    </h2>
    <div
      class="chakra-stack css-1uy6ub9"
    >
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        Buy now
      </button>
      <button
        class="chakra-button css-mg629g"
        type="button"
      >
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
